src/main/java/org/terasology/taskSystem/nui/TaskSelectionScreenLayer.java

```diff
@@ -20,90 +20,87 @@
-        });
-
-        guardCommand.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Guard);
-        });
-
-        hospitalButton.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.Hospital);
-        });
-
-        dinerButton.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.Diner);
-        });
-
-        gymButton.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.Gym);
-        });
-
-        classroomButton.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.Classroom);
-        });
-
-        storageButton.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.Storage);
-        });
-
-        laboratoryButton.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.Laboratory);
-        });
+        populateTasksList();
+        populateBuildingsList();
 
         cancelButton.subscribe(button -> {
             sendSetTaskTypeEvent();
         });
     }
 
-    public void sendSetTaskTypeEvent () {
+    private void sendSetTaskTypeEvent () {
         localPlayer.getCharacterEntity().send(new SetTaskTypeEvent());
     }
 
-    public void sendSetTaskTypeEvent(String assignedTaskType) {
+    private void sendSetTaskTypeEvent(String assignedTaskType) {
         localPlayer.getCharacterEntity().send(new SetTaskTypeEvent(assignedTaskType));
     }
 
-    public void sendSetTaskTypeEvent (String assignedTaskType, BuildingType buildingType) {
+    private void sendSetTaskTypeEvent (String assignedTaskType, BuildingType buildingType) {
         localPlayer.getCharacterEntity().send(new SetTaskTypeEvent(assignedTaskType, buildingType));
     }
+
+    private void populateBuildingsList() {
+        buildingsList = find(Constants.BUILDINGS_LIST_ID, UIList.class);
+        buildingsList.subscribeSelection(new ItemSelectEventListener<String>() {
+            @Override
+            public void onItemSelected(UIWidget widget, String item) {
+                sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.valueOf(item));
+            }
+        });
+
+        List<String> buttonList = new ArrayList<>();
+        buttonList.add("Diner");
+        buttonList.add("Storage");
+        buttonList.add("Laboratory");
+        buttonList.add("Hospital");
+        buildingsList.setList(buttonList);
+    }
+
+    private void populateTasksList() {
+        tasksList = find(Constants.TASKS_LIST_ID, UIList.class);
+
+        tasksList.subscribeSelection(new ItemSelectEventListener<String>() {
+            @Override
+            public void onItemSelected(UIWidget widget, String item) {
+                String task = AssignedTaskType.None;
+                switch(item) {
+                    case "Plant" : task = AssignedTaskType.Plant;
```

Comment:
I feel this would be more legible if it were on two lines

Reply:
Additionally if you're only checking for a single task type, then an if statement is probably more appropriate.
If you're going to be checking for a number of tasks, then there is actually a method present in the Enum class to convert a string into an Enum value.
I can see you use it in the Building task code

Reply:
`AssignedTaskType` is not an Enum it is just a collection of final stringsm since this is used in the BTs to select the appropriate behavior for a task type and BTs cannot handle assertions for enum.